### PR TITLE
feat: Add IndexTable

### PR DESCRIPTION
**Pull request by ninariccimarie**

### CODE REVIEW

File: test/e2e/indexTable/indexTablePage.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  readonly tableRow: Locator;
+  readonly moveToAction: Locator;
+
+  constructor(page: Page) {
+    super(page, '?path=/story/next-indextable--interactive');
+    this.container = page.locator('#storybook-preview-wrapper');
+    this.checkboxes = page
+      .frameLocator('internal:attr=[title="storybook-preview-iframe"i]')
+      .getByRole('checkbox');
+    this.tableRow = page
```
Comment (marthayaputra) [inline test/e2e/indexTable/indexTablePage.ts:17]: I think this one should be done as checkboxes, return list and let the test to decide which row they wanna use

File: src/@next/IndexTable/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export * as IndexTable from './IndexTable';
```
Comment (marthayaputra) [inline src/@next/IndexTable/index.ts]: missing export in `src/@next/index`?  Any reason to use `export as` here?

File: src/@next/IndexTable/components/BulkActions/BulkActions.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        active={active}
+        activator={activator}
+        onClose={onClose}
+      >
+        <Popover.Pane>
+          <div style={{ padding: '10px' }}>
+            <ButtonGroup>
+              <Button>Action</Button>
+              <Button icon={Suffix} iconPosition="right">
+                Move to
```
Comment (marthayaputra) [inline src/@next/IndexTable/components/BulkActions/BulkActions.tsx]: I think this label should be a prop

**Review by marthayaputra — APPROVED** [review]

File: src/@next/IndexTable/components/BulkActions/BulkActions.tsx (hunk 3 above)
Comment (ninariccimarie) [reply src/@next/IndexTable/components/BulkActions/BulkActions.tsx]: this component is removed in favor of using polaris. thanks for spotting

File: src/@next/IndexTable/index.ts (hunk 2 above)
Comment (ninariccimarie) [reply src/@next/IndexTable/index.ts]: my bad. i removed the `as` and also added the export

File: test/e2e/indexTable/indexTablePage.ts (hunk 1 above)
Comment (ninariccimarie) [reply test/e2e/indexTable/indexTablePage.ts:17]: i wasn't able to find a way to do it the same as checkboxes as there wasn't a `getByRole('tr')` or `getByRole('td')`. do you know any method?

**Pull request merged**